custom/icds_reports/sqldata/exports/dashboard_usage.py

```diff
@@ -0,0 +1,209 @@
+        '.bpa': 'Block Project Assistant',
+        '.dpo': 'District Programme Officer (DPO)',
+        '.cdpo': 'Child Development Project Officer (CDPO)',
+
+    }
+
+    def __init__(self, couch_user, domain):
+        self.user = couch_user
+        self.domain = domain
+        self.agg_list = None
+        self.sql_locations = None
+
+    def get_role_from_username(self, username):
+        for key, value in self.roles.items():
+            if key.lower() in username:
+                return value
+        return 'N/A'
+
+    def get_agg_fields(self):
+        location_list = self.location_types.pop()
+        return location_list.append('awc_id')
+
+    def convert_boolean_to_string(self, value):
+        return 'yes' if value else 'no'
+
+    def get_users_by_location(self, user_supported_locations):
+        user_query = UserES().mobile_users().domain(self.domain).location(
+            user_supported_locations
+        ).fields(['username', 'assigned_location_ids', 'last_login'])
+        return [u for u in user_query.run().hits]
+
+    def get_names_from_ids(self, location_id):
+        if location_id == 'doc_id':
+            return 'awc_name'
+        else:
+            return location_id.split('_')[0] + '_name'
+
+    def convert_rs_to_matrix(self, results_queryset, main_location_type):
+        """
+
+        :param results_queryset: AwcLocation values queryset
+        :param main_location_type: parent location type to fetch only locations below that level
+        :return: location matrix similar to a dataframe and locations ids of all the locations below the given
+         location
+        """
+        location_matrix = []
+        location_ids = []
+        sub_location_types = self.location_types[self.location_types.index(main_location_type) + 1:]
+        for result in results_queryset:
+            row_data = [result['state_id'], result['district_id'], result['block_id'], result['supervisor_id'],
+                        result['doc_id']]
+            location_matrix.append(row_data)
+            # adding only descendants to the main location
+            for sub_location in sub_location_types:
+                if result[sub_location] not in location_ids:
+                    location_ids.append(result[sub_location])
+        return location_matrix, location_ids
+
+    def get_location_id_string_from_location_type(self, location_type):
+        if location_type == 'awc':
+            location_id = 'doc_id'
+        else:
+            location_id = location_type + '_id'
+        return location_id
+
+    def get_location_type_string_from_location_id(self, location_id):
+        return location_id.replace('_id', '')
+
+    def check_if_date_in_last_week(self, date):
+        if date is None:
+            return False
+        d = datetime.datetime.strptime(date, "%Y-%m-%d")
+        now = datetime.datetime.now()
+        return (d - now).days < 7
+
+    def prepare_is_launched_agg_list(self, location_type, location_id):
```

Comment:
could you brief about the intent of this function? its unclear to me whats actually happening here

Reply:
preparing a list of location_id vs is_launched and location_id vs location_type mapping